p2p/subscription.go

```diff
@@ -14,19 +14,22 @@ import (
 type subscription[H header.Header[H]] struct {
 	topic        *pubsub.Topic
 	subscription *pubsub.Subscription
+	metrics      *subscriberMetrics
 }
 
 // newSubscription creates a new Header event subscription
 // on the given host.
-func newSubscription[H header.Header[H]](topic *pubsub.Topic) (*subscription[H], error) {
+func newSubscription[H header.Header[H]](topic *pubsub.Topic, metrics *subscriberMetrics) (*subscription[H], error) {
 	sub, err := topic.Subscribe()
 	if err != nil {
 		return nil, err
 	}
+	metrics.observeSubscription(context.TODO(), 1)
```

Comment:
context?

Reply:
There is no accessible context and value of making one is minimal